dotCMS/src/main/java/com/dotcms/workflow/helper/WorkflowHelper.java

```diff
@@ -0,0 +1,519 @@
+        final String newid = id.substring
+                (id.indexOf("-") + 1, id.length());
+
+        if(id.startsWith("user-")) {
+
+            role = this.roleAPI.loadRoleByKey(newid);
+        } else if(id.startsWith("role-")) {
+
+            role = this.roleAPI.loadRoleById (newid);
+        } else {
+
+            role = this.roleAPI.loadRoleById (id);
+        }
+
+        return role;
+    } // resolveRole.
+
+    /**
+     * Finds the actions by scheme
+     * @param schemeId String
+     * @param user     User
+     * @return List of WorkflowAction
+     */
+    public List<WorkflowAction> findActionsByScheme(final String schemeId,
+                                                    final User user) {
+
+        List<WorkflowAction> actions = null;
+        final WorkflowScheme workflowSchemeProxy = new WorkflowScheme();
+
+        try {
+
+            workflowSchemeProxy.setId(schemeId);
+            actions =
+                    this.workflowAPI.findActions(workflowSchemeProxy, (null == user)?
+                            APILocator.getUserAPI().getSystemUser(): user);
+        } catch (DotDataException | DotSecurityException e) {
+
+            Logger.error(this.getClass(), e.getMessage(), e);
+            throw new DotWorkflowException(e.getMessage(), e);
+        }
+
+        return actions;
+    } // findActionsByScheme.
+
+    /**
+     * Save a WorkflowActionForm returning the WorkflowAction created.
+     * A WorkflowActionForm can send a stepId in that case the Action will be associated to the Step in the same transaction.
+     * @param workflowActionForm WorkflowActionForm
+     * @return WorkflowAction (workflow action created)
+     */
+    @WrapInTransaction
+    public WorkflowAction save (final WorkflowActionForm workflowActionForm, final User user) {
+
+        String actionNextAssign     = workflowActionForm.getActionNextAssign();
+        if (actionNextAssign != null && actionNextAssign.startsWith("role-")) {
+            actionNextAssign  = actionNextAssign.replaceAll("role-", StringPool.BLANK);
+        }
+
+        final WorkflowHelper.IsNewAndCloneItResult isNewAndCloneItResult =
+                this.isNewAndCloneIt(workflowActionForm.getActionId());
+        final WorkflowAction newAction = isNewAndCloneItResult.getAction();
+        final boolean isNew            = isNewAndCloneItResult.isNew();
+
+        newAction.setName       (workflowActionForm.getActionName());
+        newAction.setAssignable (workflowActionForm.isActionAssignable());
+        newAction.setCommentable(workflowActionForm.isActionCommentable());
+        newAction.setIcon       (workflowActionForm.getActionIcon());
+        newAction.setNextStep   (workflowActionForm.getActionNextStep());
+        newAction.setSchemeId   (workflowActionForm.getSchemeId());
+        newAction.setCondition  (workflowActionForm.getActionCondition());
+        newAction.setRequiresCheckout(workflowActionForm.isRequiresCheckout());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of "setRequiresCheckout"; it is deprecated. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)